Go/Go.sublime-syntax

```diff
@@ -967,175 +1059,269 @@ contexts:
 
+  pop-after-type:
+    - match: \|{{operator_break}}
+      scope: keyword.operator.go
+      set: pop-type
+    - include: pop-before-newline
     - include: pop-before-nonblank
 
   pop-type-nested-in-parens:
-    - match: \)
-      scope: punctuation.section.parens.end.go
-      pop: true
+    - include: pop-on-paren-end
     - include: pop-type
 
   pop-struct:
     - match: \bstruct\b
       scope: keyword.declaration.struct.go
-      set:
-        - include: match-comments
-        - match: \{
-          scope: punctuation.section.braces.begin.go
-          set:
-            - meta_scope: meta.type.go
+      set: pop-struct-block
+
+  pop-struct-block:
+    - match: \{
+      scope: punctuation.section.braces.begin.go
+      set: [meta-type, pop-struct-block-inner]
+    - include: pop-before-nonblank
 
-            - match: \}
-              scope: punctuation.section.braces.end.go
-              pop: true
+  pop-struct-block-inner:
+    - include: pop-on-brace-end
+    - include: match-keywords
+    - include: match-type-operators
+    - include: match-namespace
 
-            - include: match-keywords
-            - include: match-star
+    - match: '{{ident_blank}}(?={{noise}}{{struct_embed_argument_list}})'
+      scope: variable.language.blank.go
+      push: pop-type-argument-list
 
-            - match: '{{ident}}(?={{noise}}\.)'
-              scope: variable.other.go
-            - match: \.
-              scope: punctuation.accessor.dot.go
-            - match: '{{predeclared_type}}(?={{noise}}(?:"|`|//|;|\}|$))'
-              scope: entity.other.inherited-class.go support.type.builtin.go
-            - match: '{{ident}}(?={{noise}}(?:"|`|//|;|\}|$))'
-              scope: entity.other.inherited-class.go
+    - match: '{{ident}}(?={{noise}}{{struct_embed_argument_list}})'
+      scope: entity.other.inherited-class.go
+      push: pop-type-argument-list
 
-            - match: \b_\b
-              scope: variable.language.blank.go
-            - match: '{{ident}}'
-              scope: variable.other.member.declaration.go
-              push:
-                - match: (?=\})
-                  pop: true
-                - include: pop-on-terminator
-                - include: match-comments
-                - include: pop-type
-                - include: match-any
+    - match: '{{predeclared_type}}(?={{noise}}{{struct_entry_delim}})'
+      scope: entity.other.inherited-class.go support.type.builtin.go
 
-            - include: match-any
+    - include: match-identifier-blank
 
-        - include: pop-before-nonblank
+    - match: '{{ident}}(?={{noise}}{{struct_entry_delim}})'
+      scope: entity.other.inherited-class.go
+
+    - match: '{{ident}}'
+      scope: variable.other.member.declaration.go
+      push: pop-struct-field-meta
+
+    - include: match-any
+
+  pop-struct-field-meta:
+    - include: pop-before-brace-end
+    - include: pop-before-terminator
+    - include: pop-type
+    - include: match-any
```

Comment:
I have a feeling about `pop-type` to need to be the last line as it starts with `(?=\S)`, but I may be wrong.

```suggestion
    - include: match-any
    - include: pop-before-brace-end
    - include: pop-before-terminator
    - include: pop-type
```

Reply:
The "pop-before" rules must come first, but you're right in that `match-any` wouldn't work after `pop-type`, so I'll just remove `match-any` from here. It was included for graceful degradation, like in some other places.